### DISCUSSION TITLE
Multiple Connections

### DISCUSSION THREAD

**Discussion by FredericLatour** [prisma/prisma#2281] [Q&A] [answered]
Hi, 
Is this possible to have multiple connections pointing to different db/schema in the same application?
I'm not referring to the possibility to query over multiple schema but rather the possibility to make distinct queries over different connections/db?
Thanks

**Comment by ryands17** [comment]
Hey @FredericLatour :wave: 
Currently Prisma doesn't support having multiple connections pointing to different schema/db, but you can track the request [here](https://github.com/prisma/prisma/issues/1487) 

May I know what would your use-case be for having sources for multiple schemas/db?

**Comment by FredericLatour** [comment]
@ryands17 thanks ... too bad.

The use case is the following:
One separate db/schema per project (could be whatever) with exactly the same set of tables.
One common db for common data.
There are not hundreds of projects or that would become really cumbersome (even though things can be automated).

So there is a mix of multi-tenant need (same set of tables over multiple db) and multiple database with different tables.

Just for information - this is usually easily solved by existing orm:
* TypeOrm has direct support for multiple connections (getConnection("cx1").getRepository...
* mikro-orm does not have specific support for multiple connections but you can instantiate multiple.
For both of them, you link entities to each connection so that you don't have to repeat entity definition for the situation when you have the same set of tables.

I looked at the issue you are mentioning but it doesn't look like the same matter. As far as I could understand the request in this issue is for being able to switch the type of database depending on the environment (for instance being able to use Sqlite in dev while using posgres in prod).

Looking at your schema, I believe that could be easily achieved (well kind of) on the schema side of things:

```typescript
datasource proj01 {
  provider = "postgresql"
  url      = env("DATABASE_URL_PROJ01")
  models = [Post, User]
}

datasource proj02 {
  provider = "postgresql"
  url      = env("DATABASE_URL_PROJ02")
  models = [Post, User]
}

datasource common {
  provider = "postgresql"
  url      = env("DATABASE_URL_COMMON")
  models = [Config]
}



generator client {
  provider = "prisma-client-js"
}
model Post {
  id        Int     @id @default(autoincrement())
  title     String
  content   String?
  published Boolean @default(false)
  author    User?   @relation(fields:  [authorId], references: [id])
  authorId  Int?
}
model User {
  id    Int     @id @default(autoincrement())
  email String  @unique
  name  String?
  posts Post[]
}

model Config {
  id    Int     @id @default(autoincrement())
  field string
}
```

You would then need to make it possible to use one connection/db or another:

```typescript
import { PrismaClient } from '@prisma/client'
const proj01= new PrismaClient('proj01')
const proj02= new PrismaClient('proj02')
const proj01Users = await proj01.user.findMany()
const proj02Users = await proj02.user.findMany()
```

Even for those apps that are based around one single schema, it happens here and there that there is the need to access some other database.

By the way, even though it's a bit more involved in terms of dev workflow, Prisma looks really great ... congrats.

**Comment by FredericLatour** [comment]
@ryands17 
No feedback on that matter?
The issue you pointed me to has nothing to do with the matter here.
Honestly, it's quite surprising. Any db driver or ORM I know is capable of connecting to multiple databases either by direct support or by creating 2 instances.

**Comment by ryands17** [comment, marked as answer]
Hey @FredericLatour :wave: 
The issue that I have pointed above has issues linked to them regarding multiple connections as well as supporting Read Replicas.
But as your above use-case is different, I would suggest creating a new Feature Request regarding this with the attached description above.

**Reply by FredericLatour** [reply]
@ryands17 I created the following new issue #2443 .
Thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
